src/Pretzel/Program.cs

```
@@ -113,63 +119,81 @@ public void Compose(BaseParameters parameters)
-                                {
-                                    pluginsPath,
-                                    new[]
-                                    {
-                                        typeof(DotLiquid.Tag),
-                                        typeof(Logic.Extensibility.ITag),
-                                        typeof(Logic.Templating.Context.SiteContext),
-                                        typeof(IFileSystem),
-                                        typeof(IConfiguration),
-                                    }
-                                });
-                            mainCatalog.Catalogs.Add(catalog);
+                            var asm = Assembly.LoadFrom(file);
+                            configuration.WithAssembly(asm);
                         }
-                        else
+                        catch (ReflectionTypeLoadException)
+                        {
+                        }
+                        catch (BadImageFormatException)
                         {
-                            Tracing.Debug("Assembly 'Pretzel.ScriptCs.dll' detected and loaded, type 'Pretzel.ScriptCs.ScriptCsCatalogFactory' found but method 'CreateScriptCsCatalog' not found.");
                         }
                     }
-                    else
-                    {
-                        Tracing.Debug("Assembly 'Pretzel.ScriptCs.dll' detected and loaded but type 'Pretzel.ScriptCs.ScriptCsCatalogFactory' not found.");
-                    }
-                }
-                else
-                {
-                    Tracing.Debug("Assembly 'Pretzel.ScriptCs.dll' detected but not loaded.");
+
+                    AddScriptCs(configuration, pluginsPath);
                 }
             }
         }
+
+        private void AddScriptCs(ContainerConfiguration configuration, string pluginsPath)
+        {
+            //TODO: Make ScriptCS Working
```

Since ScriptCs isn't available in .net standard for now, I think the best is to let it works only in net451 and to throw an exception in netstandard (only if the directory is found), until we replace it by [dotnet script](https://github.com/filipw/dotnet-script) or another solution.

---

It isn't supported by `scriptcs.componentmodel.composition` So it doesn't work with MEF2 at all. There is a contribution project to ScriptCS for MEF2 but it's empty. 😢

---

Sorry, I was confused. Could you just throw an exception for now and we will find a replacement in another PR.

---

Done.

---

Perfect!